src/odemis/gui/cont/tabs.py

```diff
@@ -2622,34 +2703,50 @@ def _perform_switch_position_movement(self, target_button):
         # Only proceed if there is no currently running target_position
         if self._move_future._state == RUNNING:
             return
-        stage = self.tab_data_model.main.stage
-        self._start_pos = stage.position.value
         # Get the required target_position from the pressed button
         self._target_position = next((m for m in self.position_btns.keys() if target_button == self.position_btns[m]),
                                      None)
         if self._target_position is None:
             return
-        # target_position metadata has the end positions for all movements except milling
-        if self._target_position in self.target_position_metadata.keys():
+
+        if self.tab_data_model.main.role == 'enzel': 
+            # define the start position 
+            stage = self.tab_data_model.main.stage
+            self._start_pos = stage.position.value
+            # target_position metadata has the end positions for all movements except milling
+            if self._target_position in self.target_position_metadata.keys():
+                if self._current_position is LOADING:
+                    box = wx.MessageDialog(self.main_frame, "The sample will be loaded. Please make sure that the sample is properly set and the insertion stick is removed.",
+                                        caption="Loading sample", style=wx.YES_NO | wx.ICON_QUESTION| wx.CENTER)
+
+                    box.SetYesNoLabels("&Load", "&Cancel")
+                    ans = box.ShowModal()  # Waits for the window to be closed
+                    if ans == wx.ID_NO:
+                        return
+                # Save the milling angle control current value (to return to it when switch back to milling)
+                if self._current_position is MILLING:
+                    milling_angle = self._get_milling_angle_value()
+                    if milling_angle is not None:
+                        self._prev_milling_angle = milling_angle
+                self._end_pos = self.target_position_metadata[self._target_position]
+                return cryoSwitchSamplePosition(self._target_position)
+            else:
+                # Target position is milling, get rx value from saved milling angle
+                rx_angle_value = self.ion_to_sample + self._prev_milling_angle
+                return cryoTiltSample(rx=rx_angle_value)
```

Comment:
This milling part will be removed on the updated tilted objective specs. With that in mind, performing the switch movement would be the same for enzel and meteor. I suggest you remove these lines from the function, then the code would much be simplified as:

```
self._start_pos = self.stage.position.value
 # target_position metadata has the end positions for all movements except milling
if self._current_position is LOADING:
    box = wx.MessageDialog(self.main_frame, "The sample will be loaded. Please make sure that the sample is properly set and the insertion stick is removed.",
                                    caption="Loading sample", style=wx.YES_NO | wx.ICON_QUESTION| wx.CENTER)

     box.SetYesNoLabels("&Load", "&Cancel")
     ans = box.ShowModal()  # Waits for the window to be closed
     if ans == wx.ID_NO:
          return
self._end_pos = self.target_position_metadata[self._target_position]
return cryoSwitchSamplePosition(self._target_position)
```

No need to check for enzel or meteor..